cli/pcluster/cw-dashboard-substack.cfn.yaml

```diff
@@ -0,0 +1,253 @@
+                {%- endfor %}
+                {{ update_position(6,6) }}
+              {%- endfor %}
+              {{ update_pos_after_section(6) }}
+
+              # EBS metrics graphs
+              - ',{"type":"text","x":{{ coord.x }},"y":{{ coord.y }},"width":24,"height":1,"properties":{"markdown":"\n# EBS Metrics\n"}}'
+              {{ update_pos_after_section(1) }}
+
+              {#- Unconditional EBS metrics #}
+              {%- set ebs_metrics = [["VolumeReadOps", "VolumeWriteOps"], ["VolumeReadBytes", "VolumeWriteBytes"],
+                                     ["VolumeTotalReadTime", "VolumeTotalWriteTime"], ["VolumeQueueLength"],
+                                     ["VolumeIdleTime"]] %}
+              {%- set number_of_ebs_volumes = config.NumberOfEBSVol|int -%}
+              {%- set ebs_volume_types = config.VolumeType.split(',') %}
+
+              {%- for metrics_group in ebs_metrics %}
+
+              # EBS {{ metrics_group|join(', ') }} graph
+              - !Sub ',{"type":"metric","x":{{ coord.x }},"y":{{ coord.y }},"width":6,"height":6,"properties":{"view":"timeSeries","stacked":false,"region":"${AWS::Region}","metrics":'
+                {%- set rendered_metrics_group_list = [] %} {# FIXME This would be interesting so that we could join using commas. #}
+                {# Previous line would also be useful to concatenate the lines for the different volumes together (but would it be better? #}
+                {# only one Sub for 5 variables but all characters for the different volumes would be concatenated... maybe less readable?) #}
+                {%- for i in range(number_of_ebs_volumes) %}
+                  {%- set volume_loop = loop %}
+                  {%- for metric in metrics_group %}
+              - !Sub
+                - '{% if volume_loop.first and loop.first %}[{% else %},{% endif %}["AWS/EBS","{{ metric }}","VolumeId","${EBS_Volume{{ i|int +1 }}}"]{% if volume_loop.last and loop.last %}]}}{% endif %}'
+                - EBS_Volume{{ i|int +1 }}: !Select
+                    - '{{ i|int }}'
+                    - !Ref 'EBSVolumesIDs'
+                  {%- endfor %}
+                {%- endfor %}
+                {{ update_position(6,6) }}
+              {%- endfor %}
+
+              {# Conditional EBS metrics -#}
+              {%- set ebs_metrics_conditions = [{'metric': 'VolumeConsumedReadWriteOps', 'accepted_vol_types': ["io1"]},
+                                                {'metric': 'VolumeThroughputPercentage', 'accepted_vol_types': ["io1"]},
+                                                {'metric': 'BurstBalance', 'accepted_vol_types': ["gp2", "st1", "sc1"]}]
+              %}
+              {%- for metric_condition in ebs_metrics_conditions %}
+                {%- set isPresent = {'bool': False} %}
+                {%- for i in range(number_of_ebs_volumes) if not isPresent.bool %}
```

Comment:
isPresent.bool --> 
1. let's use PEP8 snake_case. 
2. is_present is not meaningful as variable name. What about is_supported_vol_present or something like this?

Reply:
Done, is_supported_vol_present is nice